docs/operators/deduper.md

```diff
@@ -0,0 +1,526 @@
+Deduper - Operator Documentation
+=============
+
+Introduction
+=========================
+
+About this document
+--------------------------------
+
+This document is intended as a guide for understanding and using
+the Dedup operator. It will guide the reader on the functionality and
+usage of Dedup operator.
+
+Terminology
+------------------------
+
+We will refer to this operator as the Deduper or Dedup operator
+interchangeably.
+
+Overview
+=====================
+
+Dedup - “What” in a Nutshell
+-----------------------------------------
+
+Dedup is actually short for Deduplication. Duplicates are omnipresent and
+can be found in almost any kind of data. Most of the times it is
+essential to discard, or at the very least separate out the data into
+unique and duplicate components. The entire purpose of this
+operator is to de-duplicate data. In other words, when data passes
+through this operator, it will be segregated into two different data
+sets, one of which contains all unique tuples, and the other which are
+repeating more than once in the original data set.
+
+
+
+![](images/deduper/image00.png)
+
+
+Dedup - “How” in a Nutshell
+----------------------------------------
+
+In order to quickly decide whether an incoming tuple is duplicate
+or unique, it has to store each incoming tuple (or a signature, like key
+for example) to be used for comparison later. A plain in-memory storage
+for such a huge data is hardly scalable. Deduper employs a large scale
+distributed hashing mechanism (known as the Managed State) which allows
+it to identify if a particular tuple is duplicate or unique.
+Each time it identifies a tuple as a unique tuple, it also
+stores it into a persistent store called the Managed state for lookup in
```

Comment:
stores it into a persistent store called the Managed state for lookup in the future. => stores it into the Managed State for future lookup.
{Does it have to be persistent ? If so, suggest adding to the previous sentence, e.g. "distributed persistent hashing mechanism"